middle_end/compilenv.ml

```diff
@@ -253,59 +211,58 @@ let toplevel_approx :
   (string, Clambda.value_approximation) Hashtbl.t = Hashtbl.create 16
 
 let record_global_approx_toplevel () =
-  Hashtbl.add toplevel_approx current_unit.ui_name
+  Hashtbl.add toplevel_approx
+    (CU.Name.to_string (CU.name current_unit.ui_name))
     (get_clambda_approx current_unit)
 
 let global_approx id =
   if Ident.is_predef id then Clambda.Value_unknown
   else try Hashtbl.find toplevel_approx (Ident.name id)
   with Not_found ->
+    (* XXX Does this need anything to do with packs on it? *)
     match get_global_info id with
       | None -> Clambda.Value_unknown
       | Some ui -> get_clambda_approx ui
 
-(* Return the symbol used to refer to a global identifier *)
+(* Determination of pack prefixes for units and identifiers *)
 
-let symbol_for_global id =
-  if Ident.is_predef id then
-    "caml_exn_" ^ Ident.name id
-  else begin
-    let unitname = Ident.name id in
-    match
-      try ignore (Hashtbl.find toplevel_approx unitname); None
-      with Not_found -> get_global_info id
-    with
-    | None -> make_symbol ~unitname:(Ident.name id) None
-    | Some ui -> make_symbol ~unitname:ui.ui_symbol None
-  end
+let pack_prefix_for_current_unit () =
+  CU.for_pack_prefix current_unit.ui_name
 
-(* Register the approximation of the module being compiled *)
+let is_ident_in_current_unit id =
```

Comment:
I wonder if the name of this function might be misleading.  If `Hashtbl.mem ...` is true then I think it's the case that the ident is in the current unit.  However if this membership test is false, I think it might still be the case that the ident could be in the current unit -- it just hasn't been added to `toplevel_approx` yet, maybe?  The behaviour does seem to follow the old code, but we should check and clarify this somehow.

Reply:
It's only used once, in `pack_prefix_for_global_ident`, where the ident is known to be global. I'm not 100% sure in what sense a global is ever "in" the current unit, though? Is this only relevant during packing?

Reply:
Let's just inline this function into the single call site.

Reply:
Done.